source/adios2/engine/dataman/DataManReader.cpp

```diff
@@ -113,63 +116,46 @@ StepStatus DataManReader::BeginStep(StepMode stepMode,
     {
         if (m_Verbosity >= 5)
         {
-            std::cout << "DataManReader::BeginStep() returned EndOfStream, final step is " << m_FinalStep << std::endl;
+            std::cout << "DataManReader::BeginStep() returned EndOfStream, "
+                         "final step is "
+                      << m_FinalStep << std::endl;
         }
         return StepStatus::EndOfStream;
     }
 
-    format::DmvVecPtr vars = nullptr;
     auto start_time = std::chrono::system_clock::now();
 
-    while (vars == nullptr)
+    while (m_CurrentStepMetadata == nullptr)
     {
-        m_MetaDataMap = m_DataManSerializer.GetMetaData();
-
         if (m_ProvideLatest)
         {
-            size_t maxStep = 0;
-            for (const auto &i : m_MetaDataMap)
-            {
-                if (maxStep < i.first)
-                {
-                    maxStep = i.first;
-                }
-            }
-            m_CurrentStep = maxStep;
+            m_CurrentStepMetadata = m_DataManSerializer.GetLatestStep(
+                m_CurrentStep, m_TotalWriters);
         }
         else
         {
-            size_t minStep = std::numeric_limits<size_t>::max();
-            for (const auto &i : m_MetaDataMap)
-            {
-                if (minStep > i.first)
-                {
-                    minStep = i.first;
-                }
-            }
-            m_CurrentStep = minStep;
+            m_CurrentStepMetadata = m_DataManSerializer.GetEarliestStep(
+                m_CurrentStep, m_TotalWriters);
         }
 
-        auto currentStepIt = m_MetaDataMap.find(m_CurrentStep);
-        if (currentStepIt == m_MetaDataMap.end())
+        auto now_time = std::chrono::system_clock::now();
+        auto duration = std::chrono::duration_cast<std::chrono::seconds>(
+            now_time - start_time);
+        if (duration.count() < timeoutSeconds)
         {
-            auto now_time = std::chrono::system_clock::now();
-            auto duration = std::chrono::duration_cast<std::chrono::seconds>(
-                now_time - start_time);
-            if (duration.count() > timeoutSeconds)
+            if (m_Verbosity >= 5)
             {
-                return StepStatus::EndOfStream;
+                std::cout << "DataManReader::BeginStep() returned EndOfStream, "
+                             "final step is "
+                          << m_FinalStep << std::endl;
```

Comment:
How do we decide what type of information goes in what verbosity level? (This is just for my own curiosity; I'm not insinuating you've made the wrong choice here . . .)